scripts/treasury/aave_leverage.py

```diff
@@ -0,0 +1,134 @@
+"""
+  Given Token In and OUT, Leverage functions
+"""
+import requests
+import json
+from great_ape_safe import GreatApeSafe
+from brownie import Contract
+from helpers.addresses import registry
+
+multisigAddress = registry.eth.badger_wallets.dev_multisig
+
+## Signer
+safe = GreatApeSafe(multisigAddress)
+
+## Mainnet
+AAVE_LENDING_POOL = safe.contract("0x7d2768dE32b0b80b7a3454c06BdAc94A69DDc7A9")
+
+WBTC = safe.contract("0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599")
+
+USDC = safe.contract("0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48")
+
+DEPOSIT_TOKEN = WBTC ## We deposit wBTC and Buy it
+BORROW_TOKEN = USDC ## We borrow USDC and use it to buy wBTC
+
+## Returns price in ETH
+PRICE_ORACLE = safe.contract("0xA50ba011c48153De246E5192C8f9258A2ba79Ca9")
+
+## UniV2 Router
+UNIV2_ROUTER = safe.contract("0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D")
+
+
+def main():
+  """
+    Example function, setup with Deposit Amount and Leverage ratio
+  """
+  deposit(123)
+  lever_up(5_000) ## 50%
+  repay(123 + 123*.5)
+
+def deposit(amount):
+  """
+    Deposit into AAVE, earns basic interest, LM incentives and allows to lever
+  """
+  DEPOSIT_TOKEN.approve(AAVE_LENDING_POOL, amount)
+  AAVE_LENDING_POOL.deposit(DEPOSIT_TOKEN, amount, safe.address, 0)
+
+
+
+def lever_up(percent_bps):
+  """
+    Given total borrow amounts, we will use `percent_bps` of it to borrow BORROW_TOKEN
+  """
+  ## Get total borrowable
+  info = AAVE_LENDING_POOL.getUserAccountData(safe.address)
+  print("info")
+  print(info)
+
+  available_borrows_eth = info[2]
+  print("available_borrows_eth")
+  print(available_borrows_eth)
+
+  ## From ETH to USDC
+  usdc_to_eth = PRICE_ORACLE.getAssetPrice(BORROW_TOKEN)
+
+  ## Calculate USDC we can borrow
+  usdc_we_can_borrow = ((available_borrows_eth * 10**(BORROW_TOKEN.decimals())) // usdc_to_eth) * percent_bps // 10_000
```

Comment:
do we need to set an upper limit for `percent_bps` and apply some check before borrow it from Aave? say, never exceed the `70%` of the maximum available to borrow, i.e., `percent_bps < 7_000`?

Reply:
this is missing a division by the ltv, since `available_borrows_eth` is already reduced by the ltv. in this state 1btc levered by 5000bps will result in 1.35btc, since the ltv is 70%.

eg if we want to add a lever of .5x with a ltv of .7, we need `borrowable * .5 / .7`